packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -72,4 +79,45 @@ export default class Proxy {
     patch(proxy, { address, authenticationToken, name })
     return this._db.update(proxy).then(extractProperties)
   }
+
+  async callProxyMethod(id, method, params, expectStream = false) {
+    const proxy = await this.getProxy(id)
+
+    const { httpRequest } = this._app
+    const response = await httpRequest(proxy.address, {
+      body: format.request(0, method, params),
+      headers: {
+        'Content-Type': 'application/json',
+        Cookie: `authenticationToken=${proxy.authenticationToken}`,
+      },
+      method: 'POST',
+      pathname: '/api',
+    })
+
+    const authenticationToken = ifDef(response.headers['set-cookie'], cookies =>
+      cookie.parse(cookies.join('; '))
```

Comment:
Why `join`?

Reply:
`cookie.parse` not support arrays

Reply:
Why would it be an array?

Reply:
Please see this [example](https://repl.it/@BadrAzizbi/cookies)